### PR TITLE
fix #3409 feat(nimbus): support ios

### LINKED ISSUE #3409: Support iOS versions
We should support mobile versions in the versions drop-down. Some investigation is needed to determine what the values should be for each application

Note: Fenix shares the same version list with Desktop.  iOS will require its own version list.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-267)
┆Issue Number: EXP-267

### PR DESCRIPTION
**Pull request by jaredlockhart**
Because

* We want to add iOS as an application to Nimbus

This commit

* Refactors all the application constants into an ApplicationConfig so it's easier to see all the parameters of an application in one place
* Adds iOS as an application
* Adds versions that cover both iOS and fenix/desktop version ranges

Example JSON DTO:

```json
{
  "schemaVersion": "1.4.0",
  "slug": "ios-experiment",
  "id": "ios-experiment",
  "arguments": {},
  "application": "org.mozilla.ios.FirefoxBeta",
  "appName": "firefox_ios",
  "appId": "org.mozilla.ios.FirefoxBeta",
  "channel": "beta",
  "userFacingName": "ios experiment",
  "userFacingDescription": "asdf",
  "isEnrollmentPaused": false,
  "bucketConfig": {
    "namespace": "ios-experiment-1",
    "start": 0,
    "count": 1000,
    "total": 10000
  },
  "probeSets": [],
  "outcomes": [],
  "branches": [
    {
      "slug": "t",
      "ratio": 1,
      "feature": {
        "featureId": "universal-bandwidth-monitored-contingency",
        "enabled": true,
        "value": {
          "plz": "tnx"
        }
      }
    },
    {
      "slug": "c",
      "ratio": 1,
      "feature": {
        "featureId": "universal-bandwidth-monitored-contingency",
        "enabled": false,
        "value": null
      }
    }
  ],
  "targeting": "localeLanguageCode == 'en'",
  "startDate": null,
  "endDate": null,
  "proposedDuration": 28,
  "proposedEnrollment": 7,
  "referenceBranch": "c",
  "featureIds": [
    "universal-bandwidth-monitored-contingency"
  ],
  "last_modified": 1617206477979
}
```

### CODE REVIEW

File: app/experimenter/experiments/constants/nimbus.py [hunk 1]
```diff
@@ -92,6 +93,64 @@ class NimbusTargetingConfig:
...
+class Channel(models.TextChoices):
+    NO_CHANNEL = ""
+    UNBRANDED = "default"
+    NIGHTLY = "nightly"
+    BETA = "beta"
+    RELEASE = "release"
+
+
+@dataclass
+class ApplicationConfig:
```
Comment (lmorchard) [inline app/experimenter/experiments/constants/nimbus.py:105]: Ah, yup, this looks like a nice inversion of the separate little constants into better related records

File: app/experimenter/kinto/tasks.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ def nimbus_check_kinto_push_queue():
     A scheduled task that passes each application to a new scheduled
     task for working with kinto
     """
-    for application in NimbusExperiment.KINTO_APPLICATION_COLLECTION:
+    for application in NimbusExperiment.Application:
```
Comment (lmorchard) [inline app/experimenter/kinto/tasks.py:32]: Cool, this seems cleaner

File: app/experimenter/nimbus-ui/schema.graphql [hunk 3]
```diff
@@ -140,6 +140,7 @@ type NimbusDocumentationLinkType {
 enum NimbusExperimentApplication {
   DESKTOP
   FENIX
+  IOS
```
Comment (lmorchard) [inline app/experimenter/nimbus-ui/schema.graphql:143]: Nice refactor, minimal impact into the GQL boundary, 👍

**Review by lmorchard — APPROVED** [review]
LGTM!

File: app/experimenter/nimbus-ui/schema.graphql (hunk 3 above)
Comment (jaredlockhart) [reply app/experimenter/nimbus-ui/schema.graphql:143]: 😈 😈 😈 😈

**Pull request merged**